### ISSUE TITLE
Broken link for Developer Guide

### ISSUE THREAD

**Issue by tim-finnigan** [aws/aws-sdk-js-v3#4856]
### Describe the issue

I think the new JS docs look great — one issue I found is with this Developer Guide link:
![image](https://github.com/aws/aws-sdk-js-v3/assets/87778557/a5d22e65-e781-4f56-ad25-72594c67dcd9)

If opened in a new tab then it will redirect to the appropriate page (https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/welcome.html) but clicking directly will redirect to [this broken link](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/preview/sdk-for-javascript/v3/developer-guide/index.html).

I found this after someone mentioned on [this reddit post](https://www.reddit.com/r/aws/comments/14djvve/comment/joqiato/?utm_source=share&utm_medium=web3x&utm_name=web3xcss&utm_term=1&utm_content=share_button) that there were broken links in the new JS docs (there may be other broken links that I'm not aware of).

### Links

https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/preview/

**Comment by MYoung25** [comment]
Thanks for taking the time to let us know!  I believe the _majority_ of broken links are related to https://github.com/aws/aws-sdk-js-v3/issues/4794.  Which was fixed in a recent release.

I have confirmed the developer guide link is broken, but it is unrelated to #4794.  I expect the broken developer guide link to be fixed in today's release, I'll keep the issue open until I confirm the fix is live.

Best,

Michael
